File: Learnify.Service/Implementation/QuestionService.cs
using Learnify.Domain.DTO;
using Learnify.Domain.Entities;
using Learnify.Repository.Interfaces;
using Learnify.Service.Interfaces;
using Learnify.Service.Mappers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Learnify.Service.Implementation
{
    public class QuestionService : IQuestionService
    {
        private readonly IQuestionRepository repository;

        public QuestionService(IQuestionRepository repository)
        {
            this.repository = repository;
        }

        public IQueryable<Question> GetAllQuestions()
        {
            return repository.GetAll();
        }

        public Question GetQuestion(Guid questionId)
        {
            var question = repository.Get(questionId);
            return question;
        }

        public Question GetQuestionWithAnswers(Guid questionId)
        {
            var question = repository.GetQuestionWithAnswers(questionId);
            return question;
        }

        public IQueryable<Question> GetQuestionsByQuiz(Guid quizId)
        {
            return repository.GetQuestionsByQuizId(quizId);
        }

        public async Task<List<Question>> GetQuestionsWithAnswersByQuiz(Guid quizId)
        {
            var questions = repository.GetQuestionsWithAnswersByQuizId(quizId);
            return questions;
        }

        public Task InsertQuestion(QuestionRequest questionRequest)
        {
            var question = QuestionMapper.CopyFromDto(questionRequest);
            repository.Insert(question);
            return Task.CompletedTask;
        }

        public Task UpdateQuestion(UpdateQuestion updateQuestion)
        {
            var question = QuestionMapper.CopyFromUpdateDto(updateQuestion);
            repository.Update(question);
            return Task.CompletedTask;
        }

        public Task DeleteQuestion(Guid questionId)
        {
            var question = repository.Get(questionId);
            repository.Delete(question);
            return Task.CompletedTask;
        }
    }
}

File: Learnify.UI/src/services/QuestionService.js
import { axios_config } from '../axios/axios_config';

const endpoint = '/questions';

export const getAllQuestions = async () => {
    try {
        const response = await axios_config.get(endpoint);

        return response.status === 200 ? response.data : [];
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getQuestionById = async (questionId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${questionId}`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getQuestionWithAnswers = async (questionId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${questionId}/answers`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getQuestionsByQuiz = async (quizId) => {
    try {
        const response = await axios_config.get(`${endpoint}/quiz/${quizId}`);

        return response.status === 200 ? response.data : [];
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getQuestionsWithAnswersByQuiz = async (quizId) => {
    try {
        const response = await axios_config.get(`${endpoint}/quiz/${quizId}/withanswers`);
       
        return response.status === 200 ? response.data : [];
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const insertQuestion = async (questionRequest) => {
    try {
        const response = await axios_config.post(endpoint, questionRequest);
       
        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const updateQuestion = async (updateQuestion) => {
    try {
        const response = await axios_config.put(endpoint, updateQuestion);
        
        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const deleteQuestion = async (questionId) => {
    try {
        const response = await axios_config.delete(`${endpoint}/${questionId}`);
        
        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};
